src/components/LoginButton.vue

```diff
@@ -0,0 +1,49 @@
+<template>
+  <button v-if="!isAuth && !isLoading" @click="login" class="loginButton">
+    Login
+  </button>
+</template>
+<script>
+export default {
+  data() {
+    return {
+      isAuth: this.$auth0.isAuthenticated,
+      isLoading: this.$auth0.isLoading,
+      token: null,
+    };
+  },
+
+  methods: {
+    async login() {
+      try {
+        await this.$auth0.loginWithPopup();
+        await this.$nextTick();
+        this.token = await this.$auth0.getAccessTokenSilently();
+        this.token && this.$store.commit("updateToken", this.token);
```

Comment:
@JatinThakur14 put these both in an `if` block since you are checking `this.token` two times